src/lib/StorageFileApi.ts

```diff
@@ -288,25 +288,30 @@ export class StorageFileApi {
   }
 
   /**
-   * Retrieve URLs for assets in public buckets
+   * Retrieve URLs for assets in public buckets and encapsulates it in a return object
    *
    * @param path The file path to be downloaded, including the path and file name. For example `folder/image.png`.
    */
   getPublicUrl(
     path: string
   ): {
-    data: { publicURL: string } | null
-    error: Error | null
-    publicURL: string | null
+    data: { publicURL: string }
+    error: null
+    publicURL: string
   } {
-    try {
-      const _path = this._getFinalPath(path)
-      const publicURL = `${this.url}/object/public/${_path}`
-      const data = { publicURL }
-      return { data, error: null, publicURL }
-    } catch (error) {
-      return { data: null, error, publicURL: null }
-    }
+    const _path = this._getFinalPath(path)
+    const publicURL = `${this.url}/object/public/${_path}`
+    const data = { publicURL }
+    return { data, error: null, publicURL }
+  }
+
+  /**
+   * Retrieve URLs for assets in public buckets and returns ONLY the url as a string
+   *
+   * @param path The file path to be downloaded, including the path and file name. For example `folder/image.png`.
+   */
+  getPublicUrlAsString(path: string): string {
+    return this.getPublicUrl(path).data.publicURL
   }
```

Comment:
If we're leaving the error clause in `getPublicUrl`, this method won't work without an error check.
How painful would you find it to continue having to do the error check in your code?
Ideally, I'd like to keep the method the same for less of a breaking change, but if it's ruining the DX for you'd I'd like to know :)

Reply:
Needless to say, in the above example no such bucket nor file exist in my supabase project.

Reply:
I think either the method should check that the bucket and file exist and then throw an error if they don't, or just not return any error at all.

Reply:
I'm inclined to agree with you. Although, I like just removing the try/catch stuff and just having:
```
const { publicURL } = storage.from('my-bucket').getPublicUrl('profile.png')
```
no need for adding extra methods, then.

Let's leave this for the weekend and merge it in after #60 gets sorted :)

Reply:
I suggested adding a new method simply for backwards compatibility. I agree that replacing the current method would be a better solution.

Sounds good to me! Let's continue the discussion after #60 is finished! 👍🏻